src/tools/debugger/nodes/BreakpointNode.java

```diff
@@ -0,0 +1,57 @@
+package tools.debugger.nodes;
+
+import com.oracle.truffle.api.Assumption;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.source.SourceSection;
+
+import som.VM;
+import tools.SourceCoordinate;
+import tools.debugger.session.BreakpointEnabling;
+import tools.debugger.session.Breakpoints.MessageBreakpointType;
+
+/**
+ * Node to represent a breakpoint at the AST level.
+ * It has two possible states, enable or disable.
+ */
+public abstract class BreakpointNode extends AbstractBreakpointNode {
+    protected final BreakpointEnabling<?> breakpoint;
+
+    protected BreakpointNode(final SourceSection sourceSection, final MessageBreakpointType type) {
+      switch (type) {
+        case MESSAGE_RECEIVER_BREAKPOINT:
+          this.breakpoint = VM.getWebDebugger().getBreakpoints().
+          getReceiverBreakpoint(SourceCoordinate.create(sourceSection));
+          break;
+        case PROMISE_RESOLVER_BREAKPOINT:
+          this.breakpoint = VM.getWebDebugger().getBreakpoints().
+          getPromiseResolverBreakpoint(SourceCoordinate.create(sourceSection));
+          break;
+        case PROMISE_RESOLUTION_BREAKPOINT:
+          this.breakpoint = VM.getWebDebugger().getBreakpoints().
+          getPromiseResolutionBreakpoint(SourceCoordinate.create(sourceSection));
+          break;
+        default:
+          this.breakpoint = null;
+          break;
+      }
+    }
+
+    /** Only to be used by the DisabledBreakpointNode. */
+    protected BreakpointNode() {
+      this.breakpoint = null;
```

Comment:
This is not used anymore since `DisabledBreakpoint` inherits directly from `AbstractBreakpointNode`, right?
If so, please remove.